mangaki/mangaki/admin.py

```diff
@@ -139,29 +142,97 @@ def make_sfw(self, request, queryset):
+            for work in queryset:
+                if work.id != chosen_id:
+                    for rating in work.rating_set.select_related('user'):
                         if Rating.objects.filter(user=rating.user, work__id=chosen_id).count() == 0:
+                            # No rating already exists for this work
                             rating.work_id = chosen_id
                             rating.save()
                         else:
+                            other_rating = Rating.objects.get(user=rating.user, work__id=chosen_id)
+                            if other_rating.date and rating.date and other_rating.date < rating.date:  # Other rating is not the latest given
+                                other_rating.choice = rating.choice
                             rating.delete()
-                    self.message_user(request, "%s a bien été supprimé." % obj.title)
-                    obj.delete()
+                    for staff in work.staff_set.all():
+                        if Staff.objects.filter(artist=staff.artist, role=staff.role).count() == 0:
+                            staff.work_id = chosen_id
+                            staff.save()
+                        else:
+                            staff.delete()
+                    for genre in work.genre.all():
+                        chosen_work.genre.add(genre)
+                    Trope.objects.filter(origin_id=work.id).update(origin_id=chosen_id)
+                    models_to_update = [WorkTitle, TaggedWork, Suggestion, Recommendation, Pairing, Reference, ColdStartRating]
+                    for model in models_to_update:
+                        model.objects.filter(work_id=work.id).update(work_id=chosen_id)
+                    self.message_user(request, "%s a bien été supprimé." % work.title)
+                    work.delete()
+            chosen_work.save()
             return None
-        deletable_objects = []
-        for obj in queryset:
-            deletable_objects.append(Rating.objects.filter(work=obj)[:10])
+
+        do_not_compare = ['sum_ratings', 'nb_ratings', 'nb_likes', 'nb_dislikes', 'controversy']
+        priority = {
+            'ACTION_REQUESTED': 2,
+            'OBVIOUS_VALUE': 1,
+            'INFO_ONLY': 0
+        }
+        row_color = {
+            'ACTION_REQUESTED': 'red',
+            'OBVIOUS_VALUE': 'green',
+            'INFO_ONLY': 'black'
+        }
+        merge_type = {}
+        rows = defaultdict(list)
+        for work in queryset.values():
+            for field in work:
+                rows[field].append(work[field])
+        fields_to_choose = []
+        template_rows = []
+        for field in rows:
+            choices = rows[field]
+            suggested = None
+            if field in do_not_compare:
+                merge_type[field] = 'INFO_ONLY'
+            elif all(choice == choices[0] for choice in choices):  # All equal
+                merge_type[field] = 'OBVIOUS_VALUE'
```

Comment:
Personally, I find it confusing that it's considered as an `OBVIOUS_VALUE` rather than `NO_ACTION_REQUIRED`.